### PR TITLE
Fixed inline assembly external identifier access.

### LINKED ISSUE #11734: [Sol->Yul] Unimplemented feature error
```
contract C {
function b() public {
    assembly {
        function asmfun(a, b, c) -> x, y, z {
            x := a
            y := b
            z := 7
        }
        let a1, b1, c1 := asmfun(1, 2, 3)
        mstore(0x00, a1)
        mstore(0x20, b1)
        mstore(0x40, c1)
        return (0, 0x60)
    }
}
}

$ solc --experimental-via-ir test.sol
Error: Unimplemented feature error in libsolidity/codegen/ir/IRGeneratorForStatements.cpp:224
 --> /tmp/test.sol:3:5:
```

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/11734

### CODE REVIEW

**Comment by ekpyron** [comment]
I haven't thought the change through entirely, but what happens for
```
contract C {
function f() public {
    uint x; 
    assembly {
        function g() -> x {
            x := 42
        }
        x := g()
    }
}
}
```
?
Generally, a few test cases would be nice.

**Comment by chriseth** [comment]
This is weird, I thought I had tests in this PR...

**Comment by chriseth** [comment]
Ah, the tests are here: https://github.com/ethereum/solidity/pull/11735/files

**Comment by chriseth** [comment]
Added two tests. Changelog not necessary since it only fixes an issue with the sol yul code gen.

File: test/libsolidity/syntaxTests/inlineAssembly/external_identifier_access_shadowing.sol [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+        assembly {
+            function g() -> x {
+                x := 42
+            }
+            x := g()
+        }
+    }
+}
+// ----
+// DeclarationError 6578: (123-124): Cannot access local Solidity variables from inside an inline assembly function.
```
Comment (ekpyron) [inline test/libsolidity/syntaxTests/inlineAssembly/external_identifier_access_shadowing.sol:12]: Yeah, that's of course a bit weird... but if we make the ``-> x`` a proper shadowing error later on, I guess it's fine for now.

**Review by ekpyron — APPROVED** [review]

File: test/libsolidity/syntaxTests/inlineAssembly/external_identifier_access_shadowing.sol (hunk 1 above)
Comment (chriseth) [reply test/libsolidity/syntaxTests/inlineAssembly/external_identifier_access_shadowing.sol:12]: By the way, this is not a new error.

**Comment by chriseth** [comment]
Removed "fixes https://github.com/ethereum/solidity/issues/11380 (maybe)" because I think it does not.

**Pull request merged**